src/utils.js

```diff
@@ -10,8 +10,19 @@ export function isRetryable (error) {
 }
 
 export function wait (error) {
+  const retryAfter = error.response.headers['retry-after']
+
+  let timeToWait
+  // If Retry-After is a non-negative decimal integer indicating the seconds
+  // to delay after the response is received
+  if (Number.isInteger(parseInt(retryAfter))) {
+    timeToWait = parseInt(retryAfter) * 1000
+  } else {
+    // If Retry-After is a date after which to retry
+    timeToWait = parseInt(new Date(retryAfter).getTime() - new Date(Date.now()).getTime())
+  }
```

Comment:
I think this could be simplified.

1. The `parseInt()` function already returns an integer or `NaN` based on the provided input. So using `Number.isInteger()` to check if the parsed value is an integer is redundant in this case. Instead, it could be `!Number.isNaN(parseInt(retryAfter))`. 
2. Parsing `retryAfter` twice is not performant
3. `new Date(retryAfter).getTime()` is the same as `Date.parse(retryAfter)`
4. `new Date(Date.now()).getTime()` is the same as `Date.now()`
5. The `parseInt()` for that is redundant because it will always be an integer
6. This could be simplified with a ternary operator.

With all of these improvements, it would be:

```suggestion
  const retryAfter = error.response.headers["retry-after"];
  const parsedRetryAfter = parseInt(retryAfter);
  const timeToWait = Number.isNaN(parsedRetryAfter)
    ? Date.parse(retryAfter) - Date.now()
    : parsedRetryAfter * 1000;
```

Also, your comment said:

> If Retry-After is a non-negative decimal integer indicating the seconds to delay after the response is received

There's no check to make sure it's non-negative, did you want to add that?

Reply:
@Lioness100 Great comments man! Thanks a lot for them.

This weekend I'll have some spare time to implement and test your suggestions. @compwright please hold on merging this branch until I push a fix.